README.md
# Bio-HICF-Schema
APIs for the HICF databases

[![Build Status](https://travis-ci.org/sanger-pathogens/Bio-HICF-Schema.svg?branch=master)](https://travis-ci.org/sanger-pathogens/Bio-HICF-Schema)    
[![License: GPL v3](https://img.shields.io/badge/License-GPL%20v3-brightgreen.svg)](https://github.com/sanger-pathogens/Bio-HICF-Schema/blob/master/license_file)   

## Contents
  * [Introduction](#introduction)
  * [Installation](#installation)
    * [Required dependencies](#required-dependencies)
    * [From Source](#from-source)
    * [Running the tests](#running-the-tests)
  * [Usage](#usage)
    * [add\_midas\_user](#add_midas_user)
    * [add\_multiple\_midas\_users](#add_multiple_midas_users)
    * [build\_schemata\.pl](#build_schematapl)
    * [delete\_midas\_user](#delete_midas_user)
    * [geocode\_samples\_cron](#geocode_samples_cron)
    * [load\_antimicrobials](#load_antimicrobials)
    * [load\_manifest](#load_manifest)
    * [load\_ontology](#load_ontology)
    * [load\_samples\_cron](#load_samples_cron)
    * [load\_tax\_tree](#load_tax_tree)
    * [set\_midas\_password](#set_midas_password)
  * [License](#license)
  * [Feedback/Issues](#feedbackissues)

## Introduction
Database schema module, including loading scripts.

## Installation
Bio-HICF-Schema has the following dependencies:

### Required dependencies
* [Bio-Metadata-Validator](https://github.com/sanger-pathogens/Bio-Metadata-Validator)

Details for installing Bio-HICF-Schema are provided below. If you encounter an issue when installing Bio-HICF-Schema please contact your local system administrator. If you encounter a bug please log it [here](https://github.com/sanger-pathogens/Bio-HICF-Schema/issues) or email us at path-help@sanger.ac.uk.

### From Source

Clone the repository:   
   
`git clone https://github.com/sanger-pathogens/Bio-HICF-Schema.git`   
   
Move into the directory and install all dependencies using [DistZilla](http://dzil.org/):   
  
```
cd Bio-HICF-Schema
dzil authordeps --missing | cpanm
dzil listdeps --missing | cpanm
```
  
Run the tests:   
  
`dzil test`   
If the tests pass, install Bio-HICF-Schema:   
  
`dzil install`   

### Running the tests
The test can be run with dzil from the top level directory:  
  
`dzil test`  
## Usage
Bio-HICF-Schema includes the following scripts.

### add_midas_user
```
add_midas_user [-dehnpu] [long options...]
	       -d STR --dbconfig STR        path to the database configuration file
	       -u STR --username STR        the username for the new user
	       -p[=STR] --passphrase[=STR]  set a password for the user
	       -e STR --email STR           the email address for the user
	       -n STR --name STR            the display name for the user
	       -h --help                    print usage message
```
### add_multiple_midas_users
```
add_multiple_midas_users [-dh] [long options...]
			 -d STR --dbconfig STR  path to the database configuration file
			 -h --help              print usage message
```
### build_schemata.pl
Dumps the two HICF schemata as DBIC models

### delete_midas_user
```
delete_midas_user [-dhu] [long options...]
		  -d STR --dbconfig STR  path to the database configuration file
		  -u STR --username STR  the username for the user to be deleted
		  -h --help              print usage message
```
### geocode_samples_cron
Cron script to geocode sample locations

### load_antimicrobials
```
load_antimicrobials [-ch] [long options...] <filename>
		    -c STR --config STR  path to the configuration file
		    -h --help            print usage message
```
### load_manifest
```
load_manifest [-cdh] [long options...] <filename>
	      -c STR --checklist STR  path to the checklist configuration file
	      -d STR --dbconfig STR   path to the database configuration file
	      -h --help               print usage message
```
### load_ontology
```
load_ontology [-cho] [long options...] <filename>
	      -o STR --ontology STR  name of the ontology to load
	      -c STR --config STR    path to the configuration file
	      -h --help              print usage message
```
### load_samples_cron
Cron script to load new assemblies for HICF samples

### load_tax_tree
```
load_tax_tree [-ch] [long options...]
	      -c STR --config STR  path to the configuration file
	      -h --help            print usage message
```
### set_midas_password
```
set_midas_password [-dhpu] [long options...]
		   -d STR --dbconfig STR    path to the database configuration file
		   -u STR --username STR    the username whose password will be changed
		   -p STR --passphrase STR  the new password for the user
		   -h --help                print usage message
```

## License
Bio-HICF-Schema is free software, licensed under [GPLv3](https://github.com/sanger-pathogens/Bio-HICF-Schema/blob/master/license_file).

## Feedback/Issues
Please report any issues to the [issues page](https://github.com/sanger-pathogens/Bio-HICF-Schema/issues) or email path-help@sanger.ac.uk.


CHANGELOG.md
# Change Log

## [Unreleased](https://github.com/sanger-pathogens/Bio-HICF-Schema/tree/HEAD)

**Merged pull requests:**

- Update README [\#26](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/26) ([ssjunnebo](https://github.com/ssjunnebo))
- 574328 add travis [\#25](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/25) ([ssjunnebo](https://github.com/ssjunnebo))
- update AUTHORS and add LICENSE [\#24](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/24) ([ssjunnebo](https://github.com/ssjunnebo))
- Add usage docs to "add\_multiple\_midas\_users" [\#23](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/23) ([jgtate](https://github.com/jgtate))
- Add a mechanism to delete users [\#22](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/22) ([jgtate](https://github.com/jgtate))
- Fix bug in loading of taxonomy data [\#21](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/21) ([jgtate](https://github.com/jgtate))
- Fix MIC is\_nullable and primary key [\#20](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/20) ([jgtate](https://github.com/jgtate))
- Fix AMR regex [\#19](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/19) ([jgtate](https://github.com/jgtate))
- Add a new field and rename an existing field [\#18](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/18) ([jgtate](https://github.com/jgtate))
- Fix manifest count [\#17](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/17) ([jgtate](https://github.com/jgtate))
- Fix tests that were failing on a different platform [\#16](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/16) ([jgtate](https://github.com/jgtate))
- Added AUTHORS with jt6 as author [\#14](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/14) ([aslett1](https://github.com/aslett1))
- Fix typo in definitions of "unknown" [\#13](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/13) ([jgtate](https://github.com/jgtate))
- Add API methods [\#12](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/12) ([jgtate](https://github.com/jgtate))
- Fix loading [\#11](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/11) ([jgtate](https://github.com/jgtate))
- Tidy tests [\#10](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/10) ([jgtate](https://github.com/jgtate))
- Add new API methods [\#9](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/9) ([jgtate](https://github.com/jgtate))
- Schema changes to handle newly split database [\#8](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/8) ([jgtate](https://github.com/jgtate))
- Fix test to handle changes to validator code [\#6](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/6) ([jgtate](https://github.com/jgtate))
- Changes to use Bio::Metadata::Types library [\#5](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/5) ([jgtate](https://github.com/jgtate))
- Hash the user's API key as well as their passphase [\#4](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/4) ([jgtate](https://github.com/jgtate))
- Filter "unknown" when finding un-geocoded locations [\#3](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/3) ([jgtate](https://github.com/jgtate))
- Geocoder [\#2](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/2) ([jgtate](https://github.com/jgtate))
- Sample loader [\#1](https://github.com/sanger-pathogens/Bio-HICF-Schema/pull/1) ([jgtate](https://github.com/jgtate))




lib/Bio/HICF/Schema/Result/Run.pm
use utf8;
package Bio::HICF::Schema::Result::Run;

# Created by DBIx::Class::Schema::Loader
# DO NOT MODIFY THE FIRST PART OF THIS FILE

=head1 NAME

Bio::HICF::Schema::Result::Run

=cut

use strict;
use warnings;

use Moose;
use MooseX::NonMoose;
use MooseX::MarkAsMethods autoclean => 1;
extends 'DBIx::Class::Core';

=head1 COMPONENTS LOADED

=over 4

=item * L<DBIx::Class::InflateColumn::DateTime>

=item * L<DBIx::Class::TimeStamp>

=item * L<DBIx::Class::PassphraseColumn>

=back

=cut

__PACKAGE__->load_components("InflateColumn::DateTime", "TimeStamp", "PassphraseColumn");

=head1 TABLE: C<run>

=cut

__PACKAGE__->table("run");

=head1 ACCESSORS

=head2 run_id

  data_type: 'integer'
  extra: {unsigned => 1}
  is_nullable: 0

=head2 sample_id

  data_type: 'integer'
  extra: {unsigned => 1}
  is_foreign_key: 1
  is_nullable: 0

=head2 sequencing_centre

  data_type: 'varchar'
  is_nullable: 1
  size: 45

=head2 err_accession_number

  data_type: 'varchar'
  is_nullable: 1
  size: 45

=head2 global_unique_name

  data_type: 'varchar'
  is_nullable: 1
  size: 45

=head2 qc_status

  data_type: 'enum'
  extra: {list => ["pass","fail","unknown"]}
  is_nullable: 1

=head2 created_at

  data_type: 'datetime'
  datetime_undef_if_invalid: 1
  is_nullable: 0
  set_on_create: 1

=head2 updated_at

  data_type: 'datetime'
  datetime_undef_if_invalid: 1
  is_nullable: 1
  set_on_update: 1

=head2 deleted_at

  data_type: 'datetime'
  datetime_undef_if_invalid: 1
  is_nullable: 1

=cut

__PACKAGE__->add_columns(
  "run_id",
  { data_type => "integer", extra => { unsigned => 1 }, is_nullable => 0 },
  "sample_id",
  {
    data_type => "integer",
    extra => { unsigned => 1 },
    is_foreign_key => 1,
    is_nullable => 0,
  },
  "sequencing_centre",
  { data_type => "varchar", is_nullable => 1, size => 45 },
  "err_accession_number",
  { data_type => "varchar", is_nullable => 1, size => 45 },
  "global_unique_name",
  { data_type => "varchar", is_nullable => 1, size => 45 },
  "qc_status",
  {
    data_type => "enum",
    extra => { list => ["pass", "fail", "unknown"] },
    is_nullable => 1,
  },
  "created_at",
  {
    data_type => "datetime",
    datetime_undef_if_invalid => 1,
    is_nullable => 0,
    set_on_create => 1,
  },
  "updated_at",
  {
    data_type => "datetime",
    datetime_undef_if_invalid => 1,
    is_nullable => 1,
    set_on_update => 1,
  },
  "deleted_at",
  {
    data_type => "datetime",
    datetime_undef_if_invalid => 1,
    is_nullable => 1,
  },
);

=head1 PRIMARY KEY

=over 4

=item * L</run_id>

=item * L</sample_id>

=back

=cut

__PACKAGE__->set_primary_key("run_id", "sample_id");

=head1 RELATIONS

=head2 files

Type: has_many

Related object: L<Bio::HICF::Schema::Result::File>

=cut

__PACKAGE__->has_many(
  "files",
  "Bio::HICF::Schema::Result::File",
  { "foreign.run_id" => "self.run_id" },
  { cascade_copy => 0, cascade_delete => 0 },
);

=head2 sample

Type: belongs_to

Related object: L<Bio::HICF::Schema::Result::Sample>

=cut

__PACKAGE__->belongs_to(
  "sample",
  "Bio::HICF::Schema::Result::Sample",
  { sample_id => "sample_id" },
  { is_deferrable => 1, on_delete => "CASCADE", on_update => "NO ACTION" },
);


# Created by DBIx::Class::Schema::Loader v0.07042 @ 2015-02-24 13:54:25
# DO NOT MODIFY THIS OR ANYTHING ABOVE! md5sum:FWg+GAIp4gQrkPsxnjLAew


# You can replace this text with custom code or comments, and it will be preserved on regeneration
__PACKAGE__->meta->make_immutable;
1;
